### DISCUSSION TITLE
Question on Prepare

### DISCUSSION THREAD

**Discussion by rsliotta** [rancher/system-upgrade-controller#170] [Q&A] [answered]
Hello....

So I have been testing the system upgrade controller and determined their is an issue in MY environment due to the way I have rook setup.  Its a small test cluster.  

Is there a way to use the prepare block to scale some deployments and statefulsets?  Can someone provide an example?

**Comment by dweomer** [comment, marked as answer]
> Hello....
> 
> So I have been testing the system upgrade controller and determined their is an issue in MY environment due to the way I have rook setup. Its a small test cluster.
> 
> Is there a way to use the prepare block to scale some deployments and statefulsets? Can someone provide an example?

Certainly you can effect state change to your cluster in the `prepare` container which happens prior to SUC cordoning/draining the node if such is configured. Your best bet to implement this would be to adapt something like what we do for the `rancher/rke2-upgrade` and `rancher/k3s-upgrade` containers which are, in a nutshell, minimally scriptable environments that effectively include a compatible kubectl (there are other approaches but this is a good place to start).

Please see:
- https://github.com/rancher/rke2-upgrade/blob/master/scripts/upgrade.sh
- https://github.com/k3s-io/k3s-upgrade/blob/master/scripts/upgrade.sh
